ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -355,6 +355,11 @@ static int component_finalize(void) {
     }
     opal_common_ucx_wpool_free(mca_osc_ucx_component.wpool);
 
+    if (mca_osc_ucx_component.env_initialized) {
```

Comment:
I think you should be able to merge it with the other code that checks for `mca_osc_ucx_component.env_initialized` few lines above (right before the `opal_common_ucx_wpool_free`).

Reply:
oh didn't see that.  tweaking.

Reply:
@bosilca check now